### PR TITLE
Convert JIT/opt to a merged test group

### PR DESCRIPTION
**Pull request by markples**
See https://github.com/markples/utils/tree/for-PR-dotnet-runtime-85847-others for ILTransform tool.  As usual, I recommend viewing the commit list since it partitions the changes in a more readable way and paying more attention to manual changes.

In this group, I was a bit more aggressive in converting individual tests in [Fact]s and [Theory]s when needing to make manual changes. This required a change to the wrapper generator (see my comment there).

- [x] outerloop, extra-platforms, gcstress

### CODE REVIEW

File: src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs [hunk 1]
```diff
@@ -753,7 +753,7 @@ private static ImmutableArray<ITestInfo> CreateTestCases(IMethodSymbol method, L
                             // Emit diagnostic
                             continue;
                         }
-                        var argsAsCode = ImmutableArray.CreateRange(args.Select(a => a.ToCSharpString()));
+                        var argsAsCode = ImmutableArray.CreateRange(args.Select(a => a.ToCSharpString() + (a.Type!.SpecialType == SpecialType.System_Single ? "F" : "")));
```
Comment (markples) [inline src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs:756]: Without this change, a float constant in [InlineData] leads to a build failure when trying to pass a double constant for a float parameter.  This isn't a great or complete solution, but I couldn't find a good public entry point in Roslyn to do this correctly, and float cases do not work at all currently.

**Comment by markples** [comment]
This is ready @trylek @dotnet/jit-contrib

**Review by trylek — APPROVED** [review]
Looks great to me, thanks Mark!

**Pull request merged**